### PR TITLE
feat : added  CRUD functionality to the availability model

### PR DESCRIPTION
**Pull request by Kavindu-248**

### CODE REVIEW

File: project/urls.py [hunk 1]
```diff
@@ -35,9 +35,14 @@
...
     path('api/', include(router.urls)),
 
-    path('api/auth/token/', TokenObtainPairView.as_view(), name='token_obtain_pair'),
-    path('api/auth/token/refresh/', TokenRefreshView.as_view(), name='token_refresh'),
+    path('api/auth/token/', TokenObtainPairView.as_view(),
+         name='token_obtain_pair'),
+    path('api/auth/token/refresh/',
+         TokenRefreshView.as_view(), name='token_refresh'),
+
+    path('api/', include('apps.assesment.urls')),
```
Comment (Raam124) [inline project/urls.py:43]: This is not the right way to add a url for a modelviewset

File: apps/assesment/views.py [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+from apps.assesment.serializers import AvalabilitySerializer
+from rest_framework import permissions
+
+
 # Create your views here.
+
+class AvalabilityViewSet(viewsets.ModelViewSet):
+    queryset = Avalability.objects.all()
+    serializer_class = AvalabilitySerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
```
Comment (Raam124) [inline apps/assesment/views.py:14]: this API is not available for non authenticated users even the GET

File: project/urls.py (hunk 1 above)
Comment (Raam124) [reply project/urls.py:43]: check the existing one in users app or check the docs

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: checked docs and changed to IsAuthenticated

File: project/urls.py (hunk 1 above)
Comment (Kavindu-248) [reply project/urls.py]: Checked the users app to gain an idea and made the required change

File: project/urls.py [hunk 3]
```diff
@@ -25,19 +25,27 @@
...
 from apps.files.views import FileViewSet
 from apps.users.views import AuthViewSet, UserViewSet
+from apps.assesment.views import AvalabilityViewSet
+
 
 router = DefaultRouter()
 router.register('auth', AuthViewSet, basename='auth')
 router.register('users', UserViewSet, basename='users')
 router.register('files', FileViewSet, basename='files')
+router.register('avalability', AvalabilityViewSet, basename='avalability')
```
Comment (Raam124) [inline project/urls.py:35]: follow the REST API best practices. it should be `availabilities`

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: project/urls.py (hunk 3 above)
Comment (Kavindu-248) [reply project/urls.py]: Referred articles on REST API design best practice and rectified the change

File: apps/assesment/views.py [hunk 4]
```diff
@@ -1,3 +1,14 @@
...
 
+from rest_framework import viewsets
+from apps.assesment.models import Avalability
+from apps.assesment.serializers import AvalabilitySerializer
+from rest_framework import permissions
+
+
 # Create your views here.
+
+class AvalabilityViewSet(viewsets.ModelViewSet):
```
Comment (Raam124) [inline apps/assesment/views.py:11]: This will throw a null value error in the doctor id in POST

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 4 above)
Comment (Kavindu-248) [reply apps/assesment/views.py:14]: Referred the documentation to make the nescessary changes

Comment (Raam124) [reply apps/assesment/views.py:14]: @Kavindu-248 this is not the way to do to also you can't set an default doctor

**Review by Raam124 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**